v3/newrelic/config.go

```diff
@@ -397,18 +411,22 @@ type Config struct {
 type CodeLevelMetricsScope uint32
 
 // These constants specify the types of telemetry data to which we will
-// attach code level metric data. Currently TransactionCLM is defined.
-// If and when other types of telemetry data are added for Code-Level
-// Metrics collection, more constants will be added here as well.
-// AllCLM
-// means to include code level metrics everywhere currently supported.
+// attach code level metric data.
+//
+// Currently, this includes
+//    TransactionCLM            any kind of transaction
+//    AllCLM                    all kinds of telemetry data for which CLM is implemented (the default)
 //
-// A scope of 0 means "all types" as a convenience so that any zero-value
-// CodeLevelMetricsScope variable provides the default expected behavior
-// rather than turning off all code level metrics.
+// The zero value of CodeLevelMetricsScope means "all types" as a convenience so that
+// new variables of this type provide the default expected behavior
+// rather than, say, turning off all code level metrics as a 0 bit value would otherwise imply.
+// Otherwise the numeric values of these constants are not to be relied
+// upon and are subject to change. Only use the named constant identifiers in
+// your code. We do not recommend saving the raw numeric value of these constants
```

Comment:
One potential use case to consider is someone trying to configure it through, say, a config file. I see that `NEW_RELIC_CODE_LEVEL_METRICS_SCOPE` is a comma-separated list of strings, but the corresponding `codeLevelMetricsScopeLabelToValue` function is private.

Reply:
It might be better to provide a fully-functional function like `ConfigFromEnvironment()` that would read all the configuration values from a file or a `map[string]string` vlaue that had been read from a config file and set them up in the agent like `ConfigFromEnvironment()` does, saving the work of custom code to take the values read from the configuration file and do the right thing for all of them.

Reply:
I created a feature request separate from this to consider providing support for config files and whatnot, but I think it would make sense to implement `encoding.TextUnmarshaler` for `CodeLevelMetricsScope` makes sense for this PR on its own.